### PR TITLE
Enforce 500 are internal_server_error

### PR DESCRIPTION
**Pull request by spolu**
We want all our 500 to be internal_server_error and vice-versa. This PR enforces it.

### CODE REVIEW

File: connectors/src/api/slack_channels_linked_with_agent.ts [hunk 1]
```diff
@@ -131,11 +131,11 @@ const _patchSlackChannelsLinkedWithAgentHandler = async (
         req,
         res,
         {
+          status_code: 400,
```
Comment (philipperolet) [inline connectors/src/api/slack_channels_linked_with_agent.ts:134]: wouldn't that be a 500 here? not being able to join a channel for variable possible reasons?

File: front/pages/api/stripe/portal.ts [hunk 2]
```diff
@@ -70,7 +70,7 @@ async function handler(
       return apiError(req, res, {
         status_code: 500,
         api_error: {
-          type: "stripe_api_error",
+          type: "internal_server_error",
           message: "An error occured while fetching the customer portal url.",
```
Comment (philipperolet) [inline front/pages/api/stripe/portal.ts:74]:
```suggestion
          message: "Stripe API: An error occured while fetching the customer portal url.",
```

File: front/pages/api/stripe/webhook.ts [hunk 3]
```diff
@@ -255,7 +245,7 @@ async function handler(
             return apiError(req, res, {
               status_code: 500,
               api_error: {
-                type: "stripe_webhook_error",
+                type: "internal_server_error",
                 message: "Error handling checkout.session.completed.",
```
Comment (philipperolet) [inline front/pages/api/stripe/webhook.ts:249]:
```suggestion
                message: "Stripe Webhook: error handling checkout.session.completed.",
```

File: front/pages/api/stripe/webhook.ts [hunk 4]
```diff
@@ -299,7 +289,7 @@ async function handler(
               return apiError(req, res, {
                 status_code: 500,
                 api_error: {
-                  type: "stripe_webhook_error",
+                  type: "internal_server_error",
                   message:
                     "Error handling customer.subscription.deleted. Subscription not found.",
```
Comment (philipperolet) [inline front/pages/api/stripe/webhook.ts:294]:
```suggestion
                    "Stripe Webhook: Error handling customer.subscription.deleted. Subscription not found.",
```

File: front/pages/api/w/[wId]/assistant/conversations/[cId]/messages/[mId]/reactions/index.ts [hunk 5]
```diff
@@ -128,10 +128,10 @@ async function handler(
         return;
       }
       return apiError(req, res, {
-        status_code: 500,
+        status_code: 400,
         api_error: {
-          type: "message_reaction_error",
-          message: "Message reaction not created.",
+          type: "invalid_request_error",
+          message: "The mesasge you're trying to react to does not exist.",
```
Comment (philipperolet) [inline front/pages/api/w/[wId]/assistant/conversations/[cId]/messages/[mId]/reactions/index.ts:134]:
```suggestion
          message: "The message you're trying to react to does not exist.",
```

File: front/pages/api/w/[wId]/assistant/conversations/[cId]/messages/[mId]/reactions/index.ts [hunk 6]
```diff
@@ -151,10 +151,10 @@ async function handler(
         res.status(200).json({ success: true });
       }
       return apiError(req, res, {
-        status_code: 500,
+        status_code: 400,
         api_error: {
-          type: "message_reaction_error",
-          message: "Message reaction not deleted.",
+          type: "invalid_request_error",
+          message: "The mesasge you're trying to react to does not exist.",
```
Comment (philipperolet) [inline front/pages/api/w/[wId]/assistant/conversations/[cId]/messages/[mId]/reactions/index.ts:157]:
```suggestion
          message: "The message you're trying to react to does not exist.",
```

**Review by philipperolet — CHANGES_REQUESTED** [review]
:+1: LGTM but for 2 typos, a couple of added words on error messages, and a question on the status of one of the errors

**Comment by philipperolet** [comment]
papertrail: in reactions.ts, this PR also removes the 500 error on null reactions

**Comment by spolu** [comment]
(null was not possible on that reactions endpoint)

**Pull request merged**